--- src/TerrariaNetCore/Terraria/GameContent/UI/ItemRarity.cs
+++ src/Unified/Terraria/GameContent/UI/ItemRarity.cs
@@ -1,5 +_,6 @@
 using System.Collections.Generic;
 using Microsoft.Xna.Framework;
+using Terraria.Enums;
 using Terraria.ID;
 
 namespace Terraria.GameContent.UI;
@@ -22,11 +_,22 @@
 		_rarities.Add(7, Colors.RarityLime);
 		_rarities.Add(8, Colors.RarityYellow);
 		_rarities.Add(9, Colors.RarityCyan);
+
+		// Added:
+		_rarities.Add(ItemRarityID.Red, Colors.RarityDarkRed);
+		_rarities.Add(ItemRarityID.Purple, Colors.RarityDarkPurple);
 	}
 
 	public static Color GetColor(int rarity)
 	{
+		/*
 		Color result = new Color(Main.mouseTextColor, Main.mouseTextColor, Main.mouseTextColor, Main.mouseTextColor);
+		*/
+
+		// Use white directly since the only place it's used where it matters
+		// multiplies byMouseTextColor anyway.
+		Color result = Color.White;
+
 		if (_rarities.ContainsKey(rarity))
 			return _rarities[rarity];
 
