--- src/TerrariaNetCore/Terraria/MessageBuffer.cs
+++ src/Unified/Terraria/MessageBuffer.cs
@@ -23,6 +_,7 @@
 using Terraria.Social.Steam;
 using Terraria.Testing;
 using Terraria.UI;
+using Terraria.Unified.Features;
 
 namespace Terraria;
 
@@ -965,8 +_,10 @@
 				if (!WorldGen.InWorld(num153, num154, 3))
 					break;
 
+				/*
 				if (Main.tile[num153, num154] == null)
 					Main.tile[num153, num154] = new Tile();
+				*/
 
 				if (Main.netMode == 2) {
 					if (!flag13) {
@@ -1151,11 +_,13 @@
 				BitsByte bitsByte19 = (byte)0;
 				BitsByte bitsByte20 = (byte)0;
 				BitsByte bitsByte21 = (byte)0;
-				Tile tile4 = null;
+				Tile tile4 = default;
 				for (int num95 = num91; num95 < num91 + num93; num95++) {
 					for (int num96 = num92; num96 < num92 + num94; num96++) {
+						/*
 						if (Main.tile[num95, num96] == null)
 							Main.tile[num95, num96] = new Tile();
+						*/
 
 						tile4 = Main.tile[num95, num96];
 						bool flag8 = tile4.active();
@@ -1639,10 +_,19 @@
 				int num24 = reader.ReadByte();
 				string name = string.Empty;
 				if (num24 != 0) {
+					/*
 					if (num24 <= 20)
 						name = reader.ReadString();
 					else if (num24 != 255)
 						num24 = 0;
+					*/
+
+					if (num24 <= ChestNameLengthExtender.PreferredMaxChestNameLength) {
+						name = reader.ReadString();
+					}
+					else if (num24 != 255) {
+						num24 = 0;
+					}
 				}
 
 				if (Main.netMode == 1) {
@@ -2050,10 +_,12 @@
 						Netplay.Clients[whoAmI].SpamWater += 1f;
 				}
 
+				/*
 				if (Main.tile[num3, num4] == null)
 					Main.tile[num3, num4] = new Tile();
+				*/
 
-				lock (Main.tile[num3, num4]) {
+				// lock (Main.tile[num3, num4]) {
 					Main.tile[num3, num4].liquid = b2;
 					Main.tile[num3, num4].liquidType(liquidType);
 					if (Main.netMode == 2) {
@@ -2063,7 +_,7 @@
 					}
 
 					break;
-				}
+				// }
 			}
 			case 49:
 				if (Netplay.Connection.State == 6) {
